metricsbenchmark_test.go

```diff
@@ -172,3 +173,81 @@ func Benchmark_MetricsEndToEnd(b *testing.B) {
 	}
 	log.Errorf("After %d iterations avg latency %.2fms", count, float64(tTime)/float64(count))
 }
+
+func Benchmark_MetricsEndToEnd_PROMQL(b *testing.B) {
+
+	/*
+	   go test -run=Bench -bench=Benchmark_MetricsEndToEnd_PROMQL -cpuprofile cpuprofile.out -o rawsearch_cpu
+	   go tool pprof ./rawsearch_cpu cpuprofile.out
+
+	   (for mem profile)
+	   go test -run=Bench -bench=Benchmark_MetricsEndToEnd_PROMQL -benchmem -memprofile memprofile.out -o rawsearch_mem
+	   go tool pprof ./rawsearch_mem memprofile.out
+	*/
+
+	config.InitializeDefaultConfig()
+	_ = localstorage.InitLocalStorage()
+	limit.InitMemoryLimiter()
+	metrics.InitTestingConfig()
+
+	baseDir := "/Users/mgaddam/Development/SigScalr/sigDup/siglens/data/ingestnodes/sigsingle.A7nWQjn5xKWrRjkqLAXQQx/"
+	config.SetSmrBaseDirForTestOnly(baseDir)
+
+	err := meta.InitMetricsMeta()
+	if err != nil {
+		b.Fatalf("failed to initialize metrics meta")
+	}
+
+	metricMetaFileName := meta.GetLocalMetricsMetaFName()
+	log.Errorf("Using metrics meta file %s", metricMetaFileName)
+	err = query.PopulateMetricsMetadataForTheFile_TestOnly(metricMetaFileName)
+	if err != nil {
+		b.Fatalf("failed to populate metrics meta")
+	}
+
+	err = query.InitQueryNode(getMyIds, serverutils.ExtractKibanaRequests)
+	if err != nil {
+		b.Fatalf("failed to initialize metrics meta")
+	}
+	startTime := uint32(1716997213)
+	endTime := uint32(1717008013)
```

Comment:
Hardcoding unix times seems like a bad idea. What happens when this is no longer a recent time range? Maybe you should set these to be the most recent 24 hours or something, relative to when the benchmark is run